File: app/Models/Player.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Player extends Model
{
    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
        'full_name',
        'city',
        'avatar',
        'bio',
        'user_id',
    ];

    /**
     * Get the user that owns the player profile.
     */
    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function profile()
    {
        return $this->hasOne(PlayerProfile::class);
    }
}

File: app/models/player.rb
class Player < ApplicationRecord
  belongs_to :user
  has_one :profile, class_name: 'PlayerProfile', dependent: :destroy
  has_many :availabilities, class_name: 'PlayerAvailability', dependent: :destroy

  validates :full_name, presence: true
  validates :city, presence: true
end